### PR TITLE
Only publish required files

### PR DESCRIPTION
**Pull request by willfarrell**
See https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files for details. Every little bit to improve cold start times.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,10 @@
         "url": "https://github.com/sponsors/ljharb"
     },
     "main": "lib/index.js",
+    "files":[
+        "dist",
+        "lib"
+    ],
```
Comment (ljharb) [inline package.json:17]: The `files` field is dangerous; it means needed files could be accidentally omitted. The only danger of `.npmignore` is that an unneeded file is unnecessarily included.

**Review by ljharb — CHANGES_REQUESTED** [review]
All files are required - `npm explore foo && npm install && npm test` should always work.###

**Comment by ljharb** [comment]
Duplicate of #356. Duplicate of #250.

The only files that currently show up in `npm pack --dry-run` that *could* be excluded are `.github/workflows/rebase.yml` and `.github/FUNDING.yml ` - every other file is 100% necessary in the published package.

**Comment by ljharb** [comment]
Separately, "cold start time" is a runtime concern, and excluding files from the published tarball has no bearing on cold start times - only (imperceptibly) on install times.

File: package.json (hunk 1 above)
Comment (willfarrell) [reply package.json:17]: Sorry, missed the npm ignore file.

Comment (ljharb) [reply package.json:17]: np, unfortunately npm's own docs recommend "files", but it's the more dangerous approach.

**Comment by willfarrell** [comment]
Thanks for the quick reply and detailed explanation. Everyone should be bundling their code in which case it doesn't have an impact on cold start time as you stated above. However, this isn't always the case and people package up everything and ship it. Will an extra 1-3kb make a difference, of course not, but when 100s of packages have extra file, they can add up to MB which can impact cold starts in my experience.

I'm mostly trying to micro optimize for those less experienced developers that end up using [middy.js v2](https://github.com/middyjs/middy/issues/585). Thanks for taking the time to review.

**Comment by ljharb** [comment]
I'm not talking about bundling - files on disk that are never required hav e *zero* impact on start time, because they're never touched.

It is, quite simply, impossible for a cold start to be impacted by unused files, unless you're including `npm install` time, or "upload and extract the deploy artifact" in start time, neither of which is typically included in that metric.

**Pull request closed without merging**